update/process/processPending.go

```diff
@@ -0,0 +1,186 @@
+type ArgsPendingTransactionProcessor struct {
+	Accounts         state.AccountsAdapter
+	TxProcessor      process.TransactionProcessor
+	RwdTxProcessor   process.RewardTransactionProcessor
+	ScrTxProcessor   process.SmartContractResultProcessor
+	PubKeyConv       state.PubkeyConverter
+	ShardCoordinator sharding.Coordinator
+	TxCoordinator    process.TransactionCoordinator
+}
+
+type pendingProcessor struct {
+	accounts         state.AccountsAdapter
+	txProcessor      process.TransactionProcessor
+	rwdTxProcessor   process.RewardTransactionProcessor
+	scrTxProcessor   process.SmartContractResultProcessor
+	pubKeyConv       state.PubkeyConverter
+	shardCoordinator sharding.Coordinator
+}
+
+type txInfo struct {
+	hash string
+	tx   data.TransactionHandler
+}
+
+// NewPendingTransactionProcessor creates a pending transaction processor to be used after hardfork import
+func NewPendingTransactionProcessor(args ArgsPendingTransactionProcessor) (*pendingProcessor, error) {
+	if check.IfNil(args.Accounts) {
+		return nil, update.ErrNilAccounts
+	}
+	if check.IfNil(args.TxProcessor) {
+		return nil, update.ErrNilTxProcessor
+	}
+	if check.IfNil(args.ScrTxProcessor) {
+		return nil, update.ErrNilSCRProcessor
+	}
+	if check.IfNil(args.RwdTxProcessor) {
+		return nil, update.ErrNilRwdTxProcessor
+	}
+	if check.IfNil(args.PubKeyConv) {
+		return nil, update.ErrNilPubkeyConverter
+	}
+	if check.IfNil(args.ShardCoordinator) {
+		return nil, update.ErrNilShardCoordinator
+	}
+
+	return &pendingProcessor{
+		accounts:         args.Accounts,
+		txProcessor:      args.TxProcessor,
+		rwdTxProcessor:   args.RwdTxProcessor,
+		scrTxProcessor:   args.ScrTxProcessor,
+		pubKeyConv:       args.PubKeyConv,
+		shardCoordinator: args.ShardCoordinator,
+	}, nil
+}
+
+// ProcessTransactionsDstMe processes all the transactions in which destination is the current shard
+func (p *pendingProcessor) ProcessTransactionsDstMe(mapTxs map[string]data.TransactionHandler) (block.MiniBlockSlice, error) {
+	sortedTxs := make([]*txInfo, 0, len(mapTxs))
+	for hash, tx := range mapTxs {
+		sortedTxs = append(sortedTxs, &txInfo{
+			hash: hash,
+			tx:   tx,
+		})
+	}
+
+	sort.Slice(sortedTxs, func(i, j int) bool {
+		return sortedTxs[i].hash < sortedTxs[j].hash
+	})
+
+	mapMiniBlocks := make(map[string]*block.MiniBlock)
+	for _, info := range sortedTxs {
+		rcvAddress, err := p.pubKeyConv.CreateAddressFromBytes(info.tx.GetRcvAddr())
+		if err != nil {
+			continue
+		}
+
+		sndAddress, err := p.pubKeyConv.CreateAddressFromBytes(info.tx.GetSndAddr())
+		if err != nil {
+			continue
+		}
+
+		sndShardId := p.shardCoordinator.ComputeId(sndAddress)
+
+		dstShardID := p.shardCoordinator.ComputeId(rcvAddress)
+		if dstShardID != p.shardCoordinator.SelfId() {
+			continue
+		}
+
+		blockType, err := p.processSingleTransaction(info)
+		if err != nil {
+			continue
+		}
+
+		localID := fmt.Sprintf("%d%d%d", sndShardId, dstShardID, blockType)
```

Comment:
add some separators as this won't work as expected when sndShardId is 11 and dstShardID is 1
Suggestion: 
```GO
localID := fmt.Sprintf("%d_%d_%d", sndShardId, dstShardID, blockType)
```

Reply:
good catch